open-api/rest-catalog-open-api.yaml

```diff
@@ -2838,6 +2991,76 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+      properties:
+        select:
+          description: A list of the selected column names
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+          default: true
+        snapshot-id:
+          description: an int64 snapshot ID (if snapshot-range is not present); optional and defaults to the table's current snapshot
+          type: integer
+          format: int64
+        snapshot-range:
+          description: A JSON list containing exactly 2 snapshot IDs representing the start (exclusive) and end (inclusive) snapshots. This option is not allowed when `snapshot-id` is present in the request.
+          type: array
+          items:
+            type: integer
+            format: int64
+      oneOf:
+        - required: [snapshot-id]
+        - required: [snapshot-range]
+
+    PlanTableRequest:
+      type: object
+      required:
+        - select
+      properties:
+        select:
+          description: A list of the selected column names
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+          default: true
+        stats-fields:
+          description: A list of string field names for which stats should be included
+          type: array
+          items:
+            type: string
+        plan-task:
+          $ref: '#/components/schemas/PlanTask'
+        snapshot-id:
+          description: an int64 snapshot ID (if snapshot-range is not present); optional and defaults to the table's current snapshot. This option is not allowed when 'plan-task` is present in the request.
+          type: integer
+          format: int64
+
+        snapshot-range:
+          description: A JSON list containing exactly 2 snapshot IDs representing the start (exclusive) and end (inclusive) snapshots. This option is not allowed when `snapshot-id` or `plan-task` is present in the request.
+          type: array
+          items:
+            type: integer
+            format: int64
+      oneOf:
+        - required: [plan-task]
```

Comment:
I don't think that's the right usage of `oneOf` here. With `oneOf` you would refer to different schema objects, such as in https://github.com/apache/iceberg/blob/81dc9d294e4a807a185b1e1e2a071ae17d8b8f89/open-api/rest-catalog-open-api.yaml#L1819-L1824

Reply:
similar to what i wrote here https://github.com/apache/iceberg/pull/9695#discussion_r1546861228